zerver/outgoing_webhooks/generic.py

```diff
@@ -0,0 +1,35 @@
+from __future__ import absolute_import
+from typing import Any, Dict, Text, Tuple, Callable, Mapping
+import json
+from requests import Response
+from zerver.models import UserProfile
+
+from zerver.lib.outgoing_webhook import OutgoingWebhookServiceInterface
+
+class Generic(OutgoingWebhookServiceInterface):
+
+    def __init__(self, base_url, token, user_profile, service_name):
+        # type: (Text, Text, UserProfile, Text) -> None
+        self.user_profile = user_profile # type: Text
+        self.service_name = service_name # type: Text
+        self.base_url = base_url # type: Text
+        self.token = token # type: Text
+
+    def process_command(self, event):
+        # type: (Dict[str, Any]) -> Tuple[Dict[str ,Any], Dict[str, Any]]
+        rest_operation = {'method': 'POST',
+                          'relative_url_path': '',
+                          'request_kwargs': {},
+                          'base_url': self.base_url}
+        trigger_cache = {"data": event['command']}
```

Comment:
There is no specific reason, I think `request_data` would be a better name for this.